From 2802761170975b972bf44557dc2c9f7139d5a7b7 Mon Sep 17 00:00:00 2001
From: "q66" <q66@chimera-linux.org>
Date: Tue, 28 Jul 2026 00:00:00 2001
Subject: [PATCH] ppc/ppc64: disable PIE for runtime link

Using PIE results in bad relocations for ppc-assem.S / ppc64-assem.S:

ld: error: relocation R_PPC64_ADDR16_HI cannot be used against local symbol; recompile with -fPIC
>>> defined in ppc64-assem.o
>>> referenced by ppc64-assem.S (./src/runtime/ppc64-assem.S)
>>>               ppc64-assem.o:(call_into_lisp)

The affected asm is `lis reg,lra@h` / `ori reg,reg,lra@l`, where `lra` is a
local symbol defined in the same translation unit as the call site.

---
 src/runtime/Config.ppc-linux   | 2 +-
 src/runtime/Config.ppc64-linux | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/runtime/Config.ppc-linux b/src/runtime/Config.ppc-linux
index 00b35dd..ab9cd73 100644
--- a/src/runtime/Config.ppc-linux
+++ b/src/runtime/Config.ppc-linux
@@ -10,7 +10,7 @@
 # files for more information.

 CFLAGS += -m32 -std=gnu99
-LINKFLAGS += -m32 -Wl,--export-dynamic
+LINKFLAGS += -m32 -Wl,--export-dynamic -no-pie

 ASSEM_SRC = ppc-assem.S
 ARCH_SRC = ppc-arch.c
diff --git a/src/runtime/Config.ppc64-linux b/src/runtime/Config.ppc64-linux
index 8f68d26..5292bc0 100644
--- a/src/runtime/Config.ppc64-linux
+++ b/src/runtime/Config.ppc64-linux
@@ -10,7 +10,7 @@
 # files for more information.

 CFLAGS += -m64 -std=gnu99
-LINKFLAGS += -m64 -Wl,--export-dynamic
+LINKFLAGS += -m64 -Wl,--export-dynamic -no-pie

 ASSEM_SRC = ppc64-assem.S
 ARCH_SRC = ppc-arch.c
--
2.41.0

